django_ipam/base/admin.py

```diff
@@ -1,9 +1,46 @@
-from django.contrib.admin import ModelAdmin
+from ipaddress import ip_address
 
+import swapper
+from django.contrib.admin import ModelAdmin
+from django.http import HttpResponse
 
-class AbstractIpAddressAdmin(ModelAdmin):
-    pass
+Subnet = swapper.load_model("django_ipam", "Subnet")
+IpAddress = swapper.load_model("django_ipam", "IpAddress")
 
 
 class AbstractSubnetAdmin(ModelAdmin):
-    pass
+    change_form_template = "admin/django-ipam/change_form.html"
+
+    def change_view(self, request, object_id, form_url="", extra_context=None):
+        instance = Subnet.objects.get(pk=object_id)
+        total = [x for x in instance.subnet.hosts()]
+        used = 0
+        used_ip = []
+        used_ip_instance = []
+        for ipaddr in IpAddress.objects.filter().values():
+            if ip_address(ipaddr["ip_address"]) in total:
+                used += 1
+                used_ip_instance.append(ipaddr)
+                used_ip.append(ip_address(ipaddr["ip_address"]))
+
+        available = len(total) - used
+        labels = ['Used', 'Available']
+        values = [used, available]
+        extra_context = {}
+        extra_context.update({'labels': labels,
```

Comment:
you are initializing a new dict and then updating it, why is that? Wouldn't make more sense to define the dict directly without using `update`?